internal/storage/control_client_wrapper.go

```diff
@@ -82,3 +98,157 @@ func withBillingProject(controlClient StorageControlClient, billingProject strin
 	}
 	return controlClient
 }
+
+// storageControlClientWithRetry is a wrapper for an existing StorageControlClient object
+// which implements gcsfuse-level retry logic if any of the control-client call gets stalled or returns a retryable error.
+// It makes time-bound attempts to call the underlying StorageControlClient methods,
+// retrying on errors that should be retried according to gcsfuse's retry logic
+type storageControlClientWithRetry struct {
+	raw StorageControlClient
+
+	// Time-limit to attempt the first invocation of a retried call.
+	initialRetryDeadline time.Duration
+	// Maximum time-limit for any subsequent attempts.
+	maxRetryDeadline time.Duration
+	// Multiplier (expected to be > 1) to scale up the value of deadline from one attempt to the next.
+	retryMultiplier float64
+	// Total duration allowed across all the attempts.
+	totalRetryBudget time.Duration
+
+	// Backoff parameters for retries.
+	initialBackoff    time.Duration
+	maxBackoff        time.Duration
+	backoffMultiplier float64
+
+	// As an example,
+	// if initialRetryDeadline is 1 second, retryMultiplier is 2,
+	// maxRetryDeadline is 5 seconds, and totalRetryBudget is 1 minute, then
+	// first attempt will be allowed to run for 1 second. If it is not
+	// completed/failed in 1 second, it will be cancelled.
+	// Immediately, a second attempt will be made with allowance
+	// for 2 seconds (1s * 2), next attempt will
+	// be 4 seconds, but the next attempt will be allowed only 5
+	// seconds (maxRetryDeadline), not 8.
+	// All subsequent attempts will be allowed upto 5 seconds, with
+	// total attempts capped at 1 minute of duration from the start
+	// of the first attempt.
+
+	// Whether or not to enable retries for GetStorageLayout call.
+	enableRetriesOnStorageLayoutCall bool
```

Comment:
Why are doing special consideration for storageLayout? 

Is to enable by default and disable later for non-zonal bucket?